lib/MusicBrainz/Server/Data/Series.pm

```diff
@@ -208,6 +235,34 @@ sub get_entities {
     });
 }
 
+sub load_entity_count {
+    my ($self, @series) = @_;
+    return unless @series;
+
+    my @entity_types = entities_with('series');
+
+    my $query = join(' UNION ALL ',
+       map {
+           my $entity_type = $_;
+           my $series_table = "${entity_type}_series";
+           "  (SELECT series, count(*) AS count
+                FROM $series_table
+               WHERE series = any(?)
+            GROUP BY series)"
+       } @entity_types,
+    );
+
+    my @series_ids = map{ $_->id } @series;
+    my @query_params = (\@series_ids) x scalar(@entity_types);
```

Comment:
Hmm. So rather than having the one `UNION ALL` query, partition the ID lists and only run (separately) the queries that actually have relevant series, then set the counts from each run of the entity type loop? Or partition the lists and change this to `\@artist_series_ids, \@event_series_ids` etc? (can we do the last, as in, can we trust the order of queries and params to always be consistent?)